### PR TITLE
feat: mariner v2 with kata config

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/nodecustomdata.yml [hunk 1]
```diff
@@ -495,6 +495,10 @@ write_files:
         [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.untrusted.options]
           BinaryName = "/usr/bin/runc"
         {{- end}}
+        {{- if IsKata }}
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata]
```
Comment (alexeldeib) [inline parts/linux/cloud-init/nodecustomdata.yml:499]: name of runtime (class)? 

custom binary path?

File: pkg/agent/testdata/MarinerV2+Kata/CustomData [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+        default_runtime_name = "runc"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+          runtime_type = "io.containerd.runc.v2"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+          BinaryName = "/usr/bin/runc"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.untrusted]
+          runtime_type = "io.containerd.runc.v2"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.untrusted.options]
+          BinaryName = "/usr/bin/runc"
+        [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.kata]
```
Comment (alexeldeib) [inline pkg/agent/testdata/MarinerV2+Kata/CustomData:168]: note the output

File: pkg/agent/baker.go [hunk 3]
```diff
@@ -846,3 +850,8 @@ func isStandardNCv1(size string) bool {
 func areCustomCATrustCertsPopulated(config datamodel.NodeBootstrappingConfiguration) bool {
 	return config.CustomCATrustConfig != nil && len(config.CustomCATrustConfig.CustomCATrustCerts) > 0
 }
+
+func isMariner(osSku string) bool {
+	// TODO(ace): consts?
+	return osSku == "CBLMariner" || osSku == "Mariner"
```
Comment (alexeldeib) [inline pkg/agent/baker.go:856]: self-nit: make these consts and possibly move this elsewhere (where?)

File: parts/linux/cloud-init/nodecustomdata.yml (hunk 1 above)
Comment (alexeldeib) [reply parts/linux/cloud-init/nodecustomdata.yml:499]: I don't know how to test this without binaries, I guess we'll have to fix it when we get the vhd

**Review by anujmaheshwari1 — APPROVED** [review]

**Pull request merged**